### ISSUE TITLE
Regression in 3.4.1 - TypeError: no implicit conversion of String into Hash

### ISSUE THREAD

**Issue by drewB** [fnando/i18n-js#553]
I just upgraded from 3.4.0 to 3.4.1 and now get "TypeError: no implicit conversion of String into Hash" error when running `rake i18n:js:export`.  This is the case in Rails 5.1 and 5.2.

```
** Execute i18n:js:export
rake aborted!
TypeError: no implicit conversion of String into Hash
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:18:in `merge'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:18:in `block in <module:Utils>'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:20:in `merge'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:20:in `block in <module:Utils>'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:20:in `merge'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:20:in `block in <module:Utils>'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:43:in `merge'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:43:in `deep_merge'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:85:in `block (2 levels) in merge_with_fallbacks!'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/fallback_locales.rb:12:in `block in each'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/fallback_locales.rb:12:in `each'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js/fallback_locales.rb:12:in `each'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:83:in `block in merge_with_fallbacks!'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:81:in `each'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:81:in `merge_with_fallbacks!'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:65:in `block in configured_segments'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:57:in `each'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:57:in `inject'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:57:in `configured_segments'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:102:in `translation_segments'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/i18n/js.rb:45:in `export'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/i18n-js-3.4.1/lib/tasks/export.rake:5:in `block (3 levels) in <top (required)>'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/task.rb:280:in `block in execute'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/task.rb:280:in `each'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/task.rb:280:in `execute'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/task.rb:218:in `block in invoke_with_call_chain'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/2.6.0/monitor.rb:235:in `mon_synchronize'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/task.rb:198:in `invoke_with_call_chain'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/task.rb:187:in `invoke'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:160:in `invoke_task'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:116:in `each'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:116:in `block in top_level'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:125:in `run_with_threads'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:110:in `top_level'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:83:in `block in run'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:186:in `standard_exception_handling'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/lib/rake/application.rb:80:in `run'
/Users/me/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/rake-13.0.0/exe/rake:27:in `<top (required)>'
/Users/me/.rbenv/versions/2.6.4/bin/rake:23:in `load'
/Users/me/.rbenv/versions/2.6.4/bin/rake:23:in `<main>'
Tasks: TOP => i18n:js:export
```

**Comment by PikachuEXE** [comment]
@ghiculescu 
It seems #551 is causing this issue
Can you take a look?

@drewB 
Would it be possible for you to provide more details like translations?
Or repeat the export process with fewer translation entries each time until you see no error?
This is to find out what translation entries are causing this error

**Comment by annacrombie** [comment]
I also just encountered this error:
```
rake aborted!
TypeError: no implicit conversion of String into Hash
/home/lattis/.local/share/gem/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:19:in `merge'
/home/lattis/.local/share/gem/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:19:in `block in <module:Utils>'
/home/lattis/.local/share/gem/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:21:in `merge'
/home/lattis/.local/share/gem/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:21:in `block in <module:Utils>'
/home/lattis/.local/share/gem/gems/i18n-js-3.4.1/lib/i18n/js/utils.rb:21:in `merge'
...
```

It was apparently attempting to merge `"Million"` with `{:one=>"millón", :other=>"millones"}`.  These locale entries occur in the rails-i18n gem, the specific problem occurs when trying to merge [`:million` in en.yml](https://github.com/svenfuchs/rails-i18n/blob/master/rails/locale/en.yml#L175) with [`:million` in es.yml](https://github.com/svenfuchs/rails-i18n/blob/master/rails/locale/es.yml#L175).

**Comment by ghiculescu** [comment]
https://github.com/fnando/i18n-js/pull/555

**Comment by drewB** [comment]
I can confirm that also fixed the issue for me.  Thanks for being so responsive everyone!

**Comment by PikachuEXE** [comment]
Fix released in `3.4.2`
